### PR TITLE
Define a new switch that allows to connect/disconnect to the printer

### PR DESCRIPTION
**Pull request by kiddouk**
Hello,

I love your work. I wanted to be able to automate the connection & disconnection because I also control the PSU of my printer and octoprint doesnt reconnect automagically.

I can now make that happen.

### CODE REVIEW

**Comment by cmroche** [comment]
Hi, thanks for the contribution. I'll take a look through the change soon and provide feedback.

File: octoprint_homeassistant/__init__.py [hunk 1]
```diff
@@ -648,6 +657,29 @@ def _generate_device_controls(self, subscribe=False):
...
+                "name": _node_name + " Connect to printer",
+                "uniq_id": _node_id + "_CONNECT",
+                "cmd_t": "~" + self._generate_topic("controlTopic", "connect"),
+                "stat_t": self._generate_topic("hassTopic", "Connected", full=True),
+                "pl_off": "off",
+                "pl_on": "on",
+                "stat_on": "Connected",
+                "stat_off": "Disconnected",
+                "device": _config_device,
+                "ic": "mdi:power-plug",
```
Comment (cmroche) [inline octoprint_homeassistant/__init__.py:679]: So mostly this look fine, but I think "mdi:power-plug" wouldn't be good here. This has a strong association with another type of device in home automation.

`mdi:lan-connect` or `mdi:check-network` seems like the might be ok choices. I am open to other options you find as well.

**Comment by cmroche** [comment]
Just the one point of feedback about the icon, once sorted out I think this can be merged. Thanks for the contribution.

**Comment by Floppy** [comment]
This is exactly the feature I'm looking for, thanks @kiddouk! Looking forward to being able to auto-connect/disconnect from my printer based on a socket switch in HA when this gets released!

**Comment by cmroche** [comment]
@kiddouk are you able to change the icon? I think this could be merged soon once that is done.

**Pull request merged**